File: C-Tutorials/bubbleSort.c
//Bubble Sort Algorithm - C
//Author: Taseen Waseq

#include <stdio.h>

void bubbleSort(int array[], int arrSize) {

//int arrSize = sizeof(array) / sizeof(array[0]);
  int temp;

  for(int x = 0; x < arrSize; x++) {
    for(int i = 0; i < arrSize - 1; i++){
      if(array[i] > array[i + 1]) {
        temp = array[i];
        array [i] = array[i + 1];
        array [i + 1] = temp;
      }
    }
  }

}

int main() {
  int numbers[] = {5,64,32,2,3,1};
  
  int arrSize = sizeof(numbers) / sizeof(numbers[0]);

  printf("Before Sort:\n");
  
  for (int i = 0; i < arrSize; ++i) {
    printf("%d  ", numbers[i]);
  }

  bubbleSort(numbers, arrSize);
  
  printf("\nAfter Sort:\n");
  
  for (int i = 0; i < arrSize; ++i) {
    printf("%d  ", numbers[i]);
  }
}

File: Java-Tutorials/BubbleSort.java
//Bubble Sort Algorithm - Java
//Author: Taseen Waseq

class BubbleSort {

    public static void bubbleSort(int [] numbers){
        int placeholder;
        int len = numbers.length;

        for(int i =0; i < len; i++){
            for(int x = 0; x < len - 1; x++){
                if(numbers[x] < numbers[x + 1]){
                    placeholder = numbers[x];
                    numbers[x] = numbers[x + 1];
                    numbers[x + 1] = placeholder;
                }
            }
        }
    }

    public static void main( String args[] ) {
        int [] array = {5,64,32,2,3,1};
        int len = array.length;

        System.out.println("Before sorting: ");
        for(int i = 0; i<len; ++i){
            System.out.print(array[i] + " ");
        }

        bubbleSort(array);

        System.out.println("\nAfter sorting: ");
        for(int i = 0; i<len; ++i){
            System.out.print(array[i] + " ");
        }
    }
    
}
